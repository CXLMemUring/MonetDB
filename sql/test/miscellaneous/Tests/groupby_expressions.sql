start transaction;
create table "groupings" ("aa" int, "bb" int);
insert into "groupings" values (1, 1), (2, 2), (1, 3);

select cast("aa"+1 as bigint) from "groupings" group by "aa"+1;
select cast(("aa"+1) + ("bb"+1) as bigint) from "groupings" group by "aa"+1, "bb"+1;
select cast(("aa"+1) + ("bb"+1) as bigint) from "groupings" group by ("aa"+1) + ("bb"+1);

select cast(sum("aa"+1) as bigint) from "groupings" group by "aa"+1;
select cast(sum(("aa"+1) + ("bb"+2)) as bigint) from "groupings" group by "aa"+1, "bb"+2;
select cast(sum(("aa"+1) + ("bb"+2)) as bigint) from "groupings" group by ("aa"+1) + ("bb"+2);
rollback;
