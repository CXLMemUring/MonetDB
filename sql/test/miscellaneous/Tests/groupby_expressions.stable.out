stdout of test 'groupby_expressions` in directory 'sql/test/miscellaneous` itself:


# 14:21:55 >  
# 14:21:55 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=32012" "--set" "mapi_usock=/var/tmp/mtest-21747/.s.monetdb.32012" "--set" "monet_prompt=" "--forcemito" "--dbpath=/home/ferreira/repositories/MonetDB-default/BUILD/var/MonetDB/mTests_sql_test_miscellaneous" "--set" "embedded_r=yes" "--set" "embedded_py=true" "--set" "embedded_c=true"
# 14:21:55 >  

# MonetDB 5 server v11.32.0 (hg id: b62008654fe6+)
# This is an unreleased version
# Serving database 'mTests_sql_test_miscellaneous', using 8 threads
# Compiled for x86_64-pc-linux-gnu/64bit with 128bit integers
# Found 15.526 GiB available main-memory.
# Copyright (c) 1993 - July 2008 CWI.
# Copyright (c) August 2008 - 2018 MonetDB B.V., all rights reserved
# Visit https://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://wired-142.cwi.nl:32012/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-21747/.s.monetdb.32012
# MonetDB/GIS module loaded
# SQL catalog created, loading sql scripts once
# loading sql script: 09_like.sql
# loading sql script: 10_math.sql
# loading sql script: 11_times.sql
# loading sql script: 12_url.sql
# loading sql script: 13_date.sql
# loading sql script: 14_inet.sql
# loading sql script: 15_querylog.sql
# loading sql script: 16_tracelog.sql
# loading sql script: 17_temporal.sql
# loading sql script: 18_index.sql
# loading sql script: 20_vacuum.sql
# loading sql script: 21_dependency_views.sql
# loading sql script: 22_clients.sql
# loading sql script: 23_skyserver.sql
# loading sql script: 25_debug.sql
# loading sql script: 26_sysmon.sql
# loading sql script: 27_rejects.sql
# loading sql script: 39_analytics.sql
# loading sql script: 39_analytics_hge.sql
# loading sql script: 40_geom.sql
# loading sql script: 40_json.sql
# loading sql script: 40_json_hge.sql
# loading sql script: 41_md5sum.sql
# loading sql script: 45_uuid.sql
# loading sql script: 46_profiler.sql
# loading sql script: 51_sys_schema_extension.sql
# loading sql script: 60_wlcr.sql
# loading sql script: 72_fits.sql
# loading sql script: 74_netcdf.sql
# loading sql script: 75_lidar.sql
# loading sql script: 75_shp.sql
# loading sql script: 75_storagemodel.sql
# loading sql script: 80_statistics.sql
# loading sql script: 80_udf.sql
# loading sql script: 80_udf_hge.sql
# loading sql script: 85_bam.sql
# loading sql script: 90_generator.sql
# loading sql script: 90_generator_hge.sql
# loading sql script: 99_system.sql
# MonetDB/SQL module loaded
# MonetDB/Python2 module loaded
# MonetDB/R   module loaded

Ready.

# 14:21:56 >  
# 14:21:56 >  "mclient" "-lsql" "-ftest" "-tnone" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-21747" "--port=32012"
# 14:21:56 >  

#start transaction;
#create table "groupings" ("aa" int, "bb" int);
#insert into "groupings" values (1, 1), (2, 2), (1, 3);
[ 3	]
#select cast("aa"+1 as bigint) from "groupings" group by "aa"+1;
% sys.L3 # table_name
% L3 # name
% bigint # type
% 1 # length
[ 2	]
[ 3	]
#select cast(("aa"+1) + ("bb"+1) as bigint) from "groupings" group by "aa"+1, "bb"+1;
% sys.L4 # table_name
% L4 # name
% bigint # type
% 1 # length
[ 4	]
[ 6	]
[ 6	]
#select cast(("aa"+1) + ("bb"+1) as bigint) from "groupings" group by ("aa"+1) + ("bb"+1);
% sys.L3 # table_name
% L3 # name
% bigint # type
% 1 # length
[ 4	]
[ 6	]
#select cast(sum("aa"+1) as bigint) from "groupings" group by "aa"+1;
% sys.L4 # table_name
% L4 # name
% bigint # type
% 1 # length
[ 4	]
[ 3	]
#select cast(sum(("aa"+1) + ("bb"+1)) as bigint) from "groupings" group by "aa"+1, "bb"+1;
% sys.L5 # table_name
% L5 # name
% bigint # type
% 1 # length
[ 5	]
[ 7	]
[ 7	]
#select cast(sum(("aa"+1) + ("bb"+2)) as bigint) from "groupings" group by ("aa"+1) + ("bb"+2);
% sys.L4 # table_name
% L4 # name
% bigint # type
% 2 # length
[ 5	]
[ 14	]
#rollback;

# 14:21:56 >  
# 14:21:56 >  "Done."
# 14:21:56 >  

